tests/gis_tests/geos_tests/test_geos.py

```diff
@@ -1434,3 +1434,12 @@ def test_empty_point(self):
         self.assertEqual(p.transform(2774, clone=True), Point(srid=2774))
         p.transform(2774)
         self.assertEqual(p, Point(srid=2774))
+
+    def test_linestring_iter(self):
+        ls = LineString((0, 0), (1, 1))
+        it = iter(ls)
+        # Step into CoordSeq iterator.
+        next(it)
+        ls[:] = []
+        with self.assertRaises(IndexError):
```

Comment:
I'd like to leave undefined what happens here as long as it doesn't segfault, because I think that it can help to make other optimizations. Specifically I was going to try to reimplement reverted optimization by making iterator owner of the old GEOS geometry. In this case the iteration will just keep going over the old geometry.

Reply:
@sir-sigurd I've already merged this patch, but we can relax this in future optimizations.